esp/bindings/http/platform/httpsecurecontext.cpp

```diff
@@ -0,0 +1,84 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2016 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "httpsecurecontext.hpp"
+
+const char* HTTP_PROPERTY_TYPE_COOKIE = "cookie";
+
+class CHttpSecureContext : extends CEspBaseSecureContext
+{
+private:
+    CHttpRequest* m_request;
+
+protected:
+    IEspContext* queryContext() const
+    {
+        return (m_request ? m_request->queryContext() : NULL);
+    }
+
+public:
+    CHttpSecureContext(CHttpRequest* request)
+        : m_request(request)
+    {
+    }
+
+    virtual ~CHttpSecureContext()
+    {
+    }
+
+    virtual const char* getProtocol() const override
```

Comment:
I don't believe it matters. This is to distinguish one protocol implementation from another, and both "http" and "https" are implemented by the same stack.